### PR TITLE
Fix for Safari (hide button not visible)

### PR DESCRIPTION
**Pull request by p0358**
Apparently the hide button was not visible on Safari.
After some research it seems that Safari hides `position: fixed` elements if parent has overflow hidden... I kept the button itself intact (to not risk that changes will render it positioned wrongly on some screens), and instead changed parent's overflow-y to visible and came up with these changes to keep everything working and looking the same (hopefully):

* `.panel` div got overflow-y changed from `auto` to `visible` (this rendered the hide button visible on Safari, but broke scrolling)
* `.panel-content` class was created to set properties on it (position absolute, width, height and overflow-y moved there to have the scrollbar there)
* `.panel-content` has `width: calc(100% - 5px);` because we compensate for the left padding of 5 px, but we want the scrollbar to be kept maximally to the right side.
This is why I added `.panel .panel-content > div { padding-right: 5px; }` to keep the padding inside, to the left of the scrollbar. (same goes for p error element)
* `.panel-content` has `height: calc(100% - 10px);` to compensate for both top and bottom paddings of 5 px (so the elements are not cut out on the bottom)
* Close button had to be moved inside of content div to keep it positioned correctly, no matter if scrollbar is visible or not. (this also keeps the behavior of staying where it was while scrolling down; not being "sticky")

These changes fix the appearance of hide button on Safari, while keeping everything else looking the same way as before.

Should be tested on some mobile Android as well before merging.

Alive proof that it works now (the right empty margin is not there, GitHub did break the screenshot while uploading somehow...):
obraz

### CODE REVIEW

File: map.css [hunk 1]
```diff
@@ -90,6 +90,24 @@ html, body, #map {
 		margin-right: 10px;
 	}
 }
+
+.panel .panel-content {
+	overflow-y: auto;
+    position: absolute;
```
Comment (jacekkow) [inline map.css:96]: Please be consistent and use tabs for indentation (here and everywhere else).

File: map.css [hunk 2]
```diff
@@ -90,6 +90,24 @@ html, body, #map {
...
+    width: calc(100% - 5px);
```
Comment (jacekkow) [inline map.css:97]: Why won't you remove that 5px padding from parent and simply move it here? This way you would not need that `.panel .panel-content > ...` classes at all.

File: map.css [hunk 3]
```diff
@@ -90,6 +90,24 @@ html, body, #map {
...
+    height: calc(100% - 10px);
+}
+.panel .panel-content .close {
```
Comment (jacekkow) [inline map.css:100]: Why is this required? It should work as-is (provided you do not modify order of elements added).

File: map.js [hunk 4]
```diff
@@ -86,14 +86,15 @@ function Panel(element) {
 	this._hide.title = lang.action_collapse;
 	this._hide.className = 'hide';
 	this._hide.addEventListener('click', this.toggleExpanded.bind(this));
-	
-	this._close = addElementWithText(this._element, 'a', '×');
+
+	this._content = document.createElement('div');
```
Comment (jacekkow) [inline map.js:90]: I would propose to leave the content as it was (it should only contain elements passed in `show` method parameter, not some-content-and-hide-button) and create a new element, maybe `_container` (having `panel-container` class), that would wrap `_hide` and `_content`.

File: map.css [hunk 5]
```diff
@@ -37,7 +37,7 @@ html, body, #map {
 	padding: 5px;
 	border-left: 1px solid black;
 	font-size: 14px;
-	overflow-y: auto;
+	overflow-y: visible;
```
Comment (jacekkow) [inline map.css:40]: If possible, just remove the directive completely.

File: map.css [hunk 6]
```diff
@@ -90,6 +90,24 @@ html, body, #map {
...
+	overflow-y: auto;
+    position: absolute;
+    width: calc(100% - 5px);
+    height: calc(100% - 10px);
+}
+.panel .panel-content .close {
+	position: absolute;
+    right: 5px;
+}
+.panel .panel-content > div {
```
Comment (jacekkow) [inline map.css:104]: You cannot assume that the content will always be `<div>` or `<p>`.

**Comment by jacekkow** [comment]
@p0358: Would you like me trying this approach?

**Comment by p0358** [comment]
@jacekkow Sorry for late reply. Yes, you can try your approach, I will be happy to test how it works

**Comment by jacekkow** [comment]
@p0358 Sorry, I have messed up this pull-request. Could you try safari-fix branch? (https://github.com/jacekkow/mpk-ttss/tree/safari-fix)

I had to create new PR, which is here: https://github.com/jacekkow/mpk-ttss/pull/32
